### ISSUE TITLE
Do We Need Zero Training Loss After Achieving Zero Training Error? (2020)

### ISSUE THREAD

**Issue by iwa-shi** [iwa-shi/paper_memo#22]
Takashi Ishida, Ikko Yamane, Tomoya Sakai, Gang Niu, Masashi Sugiyama
https://arxiv.org/abs/2002.08709
ICML2020

## 概要
学習時のロスが０にならないようにするfloodingというregularization手法を提案

## 詳細
学習時のエラー（ロスとは別。不正解というニュアンス）が０になるまで学習するべきだという先行研究が存在するが、ロスが０になるまで学習するべきかどうかはわかっていない。floodingは、エラーを０に保ちつつロスが０にならないようにする。
モデルが十分な表現力を保つ場合、通常はfig1(a)のように、初めは学習ロスとテストロスはどちらも減少するが、少しずつ差が開いていく（[A]）。その後学習ロスはさらに下がるが、テストロスが悪化する（[B]）。最終的に[C]のように学習ロスはほぼ０に収束する。そこにfloodingを用いると、fig1(b)のように学習ロスが０にならず、テストロスが改善される。

![スクリーンショット 2020-06-02 11 02 09](https://user-images.githubusercontent.com/46859277/83471783-8cbb9080-a4c0-11ea-9dc1-dd40bd7b1b4e.jpg)

具体的には、元の目的関数がJであるとき、floodingを使った目的関数は以下の式(1)で表される。

![スクリーンショット 2020-06-02 10 25 47](https://user-images.githubusercontent.com/46859277/83469769-8971d600-a4bb-11ea-98b1-ff3748bc9037.jpg)

bをflooding levelと呼ぶ。つまりJ<bの時は勾配が上昇するような目的関数になる。これによってロスはflooding levelの周辺を行ったり来たりするようになり、０にはならない。

多クラスを例に考えると、学習用ロス・テスト用ロスとして以下のロスが考えられる。
* 学習用：クロスエントロピー
* テスト用：01ロス（出力ベクトルのargmaxが正解クラスに一致するか否か）

テスト用のロスは正解率を表す。これを最小化することが目的だが、微分不可能であるためクロスエントロピーが使われる。
ここで、以下の定理が成立する。

![スクリーンショット 2020-06-02 10 35 27](https://user-images.githubusercontent.com/46859277/83470268-cbe7e280-a4bc-11ea-81a5-b0651b215a70.jpg)

このMSE(\hat R)は(\hat R - R)^2であり、01ロスとクロスエントロピーロスの二乗誤差を表す。つまり、適切なbを選べば、floodingを使った方が本来の目的関数に近づくということを意味している。

## 実験・結果
合成データを使った実験と、MNIST・CIFARを使った実験を行った。
floodingを使うことで精度は概ね改善しており、weight decay・early stoppingと組み合わせることでさらに精度が向上した例もあった。

合成データ
![スクリーンショット 2020-06-02 10 58 02](https://user-images.githubusercontent.com/46859277/83471727-6990e100-a4c0-11ea-9dab-dd7ba30026b0.jpg)

benchmark dataset
![スクリーンショット 2020-06-02 10 58 08](https://user-images.githubusercontent.com/46859277/83471757-79a8c080-a4c0-11ea-9ae0-f0d10052475e.jpg)
